bins/stackablectl/src/cmds/stacklets.rs

```diff
@@ -72,50 +76,146 @@ async fn list_cmd(args: &StackletListArgs, common_args: &Cli) -> Result<String,
         .all_namespaces
         .then_some(common_args.operator_namespace.as_str());
 
-    let stacklets = list_stacklets(namespace, StackletListOptions::default())
+    let stacklets = list_stacklets(namespace)
         .await
         .context(StackletListSnafu {})?;
 
-    println!("{:?}", stacklets);
+    if stacklets.is_empty() {
+        return Ok("No stacklets".into());
+    }
 
     match args.output_type {
         OutputType::Plain => {
-            let mut table = Table::new();
+            // Determine if colored output will be enabled based on the provided
+            // flag and the terminal support.
+            let use_color = use_colored_output(args.use_color);
 
+            // The main table displays all installed (and discovered) stacklets
+            // and their condition.
+            let mut table = Table::new();
             table
-                .set_header(vec!["PRODUCT", "NAME", "NAMESPACE", "ENDPOINTS", "INFO"])
+                .set_header(vec!["PRODUCT", "NAME", "NAMESPACE", "CONDITIONS"])
                 .set_content_arrangement(ContentArrangement::Dynamic)
                 .load_preset(UTF8_FULL);
 
+            // The error table displays optional errors in a structured manner.
+            let mut error_table = Table::new();
+            error_table
+                .set_header(vec!["#", "MESSAGES"])
+                .set_content_arrangement(ContentArrangement::Dynamic)
+                .load_preset(NOTHING);
+
+            let mut product_index = 0;
             for (product_name, products) in stacklets {
                 for product in products {
-                    let endpoints = product
-                        .endpoints
-                        .iter()
-                        .map(|(name, url)| format!("{name}:{url}"))
-                        .collect::<Vec<String>>()
-                        .join("\n");
-
-                    let additional_information = product
-                        .additional_information
-                        .iter()
-                        .map(|(key, value)| format!("{key}:{value}"))
-                        .collect::<Vec<String>>()
-                        .join("\n");
+                    let (conditions, errors) =
+                        process_conditions(product.conditions, &mut product_index, use_color);
 
                     table.add_row(vec![
                         product_name.clone(),
                         product.name,
                         product.namespace.unwrap_or_default(),
-                        endpoints,
-                        additional_information,
+                        conditions,
                     ]);
+
+                    for error in errors {
+                        error_table.add_row(vec![product_index.to_string(), error]);
+                    }
```

Comment:
I'd have to play around with it, but I *think* this formatting will end up being rather confusing.

Reply:
Improved in 0153ed2